From bee3751ec52835e55eb2b4b6807d4629663e0b00 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Thu, 9 Aug 2018 20:48:03 +0800
Subject: [PATCH 4348/5242] MLK-18994 spi: ecspi: correct the request cs gpio
 resoures process

commit  ae7946c671ed8b44cd373cd5885c956c0367b67b from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch revert "MLK-17619: spi: ecspi: request gpio resources
before setting the value", because spi_imx_setup() will be called
several times accroding to the spc definition. So remove the gpio
request operation out from spi_imx_setup(), and back to probe(),
and move up the location, just incase set the cs gpio value before
request gpio resources.

This reverts commit aba5f6342002ffe1f13a71dbddf30ec15ee1edba.
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/spi/spi-imx.c |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/drivers/spi/spi-imx.c b/drivers/spi/spi-imx.c
index 493f4cc..210c369 100644
--- a/drivers/spi/spi-imx.c
+++ b/drivers/spi/spi-imx.c
@@ -1548,11 +1548,24 @@ static int spi_imx_probe(struct platform_device *pdev)
 				master->cs_gpios[i] = mxc_platform_info->chipselect[i];
 		}
 	} else {
-		u32 num_cs;
+		u32 num_cs, cs_gpio;
 
 		if (!of_property_read_u32(np, "num-cs", &num_cs))
 			master->num_chipselect = num_cs;
+		else if (!of_property_read_u32(np, "fsl,spi-num-chipselects", &num_cs))
+			master->num_chipselect = num_cs;
+		
 		/* If not preset, default value of 1 is used */
+		master->cs_gpios = devm_kcalloc(&master->dev,
+				master->num_chipselect, sizeof(int),
+				GFP_KERNEL);
+		if (!master->cs_gpios)
+			return -ENOMEM;
+		
+		for (i = 0; i < master->num_chipselect; i++) {
+			cs_gpio = of_get_named_gpio(np, "cs-gpios", i);
+			master->cs_gpios[i] = cs_gpio;
+		}		
 	}
 
 	spi_imx->bitbang.chipselect = spi_imx_chipselect;
-- 
1.7.9.5

